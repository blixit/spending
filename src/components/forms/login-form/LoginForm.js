import React, { useCallback, useContext, useState } from 'react';
import styled from 'styled-components';

import ThemedButton from 'ui/atoms/buttons/ThemedButton';
import FormInput from 'ui/forms/form-input/FormInput';

import { EventContext } from 'core/events/provider';
import { ReadyMutations as Mutations } from 'core/http/query';
import { writeCache } from 'core/storage/cache';

import StyledError from 'components/molecules/error/Error';

const H1 = styled.h1`
  text-align: center;
  font-weight: bold;
`;

const LoginButton = styled.div`
  display: flex;
  justify-content: flex-end;
`;

const Container = styled.div`
  flex: auto;
  border-radius: 3px;
  padding: 10px;
`;

const LoginForm = props => {
  const context = useContext(EventContext);

  const [login, setLogin] = useState(null);
  const [password, setPassword] = useState(null);
  const [errors, setErrors] = useState({});
  const [hasError, setHasError] = useState(null);

  const { router } = props;
  const { eventManager } = context;

  const onLoginChanged = useCallback(login => setLogin(login), []);
  const onPasswordChanged = useCallback(password => setPassword(password), []);
  const onSubmit = useCallback(async e => {
    e.preventDefault();
    setHasError(false);

    const { login: loginQuery } = Mutations.auth;
    const data = { login, password };

    try {
      writeCache('user', data);
      eventManager.emit('refresh:activeUser', data);

      router.history.push('/');
      // const response = await mutate({ ...loginQuery, data });
    } catch (e) {
      errors.global = loginQuery.queryError || e.message;
      setHasError(true);
      setErrors(errors);
    }
  }, [errors, login, password, eventManager, router]);


  return (
    <Container {...props} >
      <H1>Your workspace</H1>
      <FormInput
        type='text'
        label='Login'
        placeholder='Enter your login'
        value={login}
        onChange={onLoginChanged}
      />
      <FormInput
        type='password'
        label='Password'
        placeholder='Enter your password'
        value={password}
        onChange={onPasswordChanged}
      />
      <LoginButton>
        <ThemedButton bgcolor='primary' onClick={onSubmit}>Log in</ThemedButton>
      </LoginButton>
      {hasError && <StyledError>{errors.global}</StyledError>}
    </Container>
  )
};

LoginForm.displayName = 'LoginForm';

export default LoginForm;
