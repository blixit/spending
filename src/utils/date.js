/**
 * Formats a date for the backend
 * @param {Date} date the date to convert
 */
export const toBackendDate = (date) => {
  return date
    ? date.toLocaleDateString() + ' ' + date.toLocaleTimeString()
    : date;
};