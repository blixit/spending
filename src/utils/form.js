/**
* Removes the listed fields from the source
* @param {*} form The source object
* @param {*} fields The fields list
*/
export const clean = (form, fields) => {
  const result = {};
  // eslint-disable-next-line no-unused-vars
  for (const field in form) {
    if (!form.hasOwnProperty(field) || fields.includes(field)) {
      continue;
    }
    result[field] = form[field];
  }
  return result;
};
