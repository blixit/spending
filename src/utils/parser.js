export const stringify = data => {
  try {
    return JSON.stringify(data);
  } catch (error) {
    return null;
  };
}

export const parse = data => JSON.parse(data);