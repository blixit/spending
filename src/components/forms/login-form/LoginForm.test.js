import { withTheme } from 'ui/theming/test.utils';

import render from 'testing/render';

import LoginForm from './LoginForm';

describe('[components/forms/filter-form] LoginForm', () => {
  it('should render properly', () => {
    const loginForm = render(withTheme(LoginForm, {}));
    expect(loginForm.toJSON()).toMatchSnapshot();
  });
});
