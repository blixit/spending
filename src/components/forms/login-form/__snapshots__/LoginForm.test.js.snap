// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[components/forms/filter-form] LoginForm should render properly 1`] = `
<div
  className="sc-dnqmqq iSJnfr"
>
  <h1
    className="sc-gzVnrw dOEsii"
  >
    Your workspace
  </h1>
  <div
    className="sc-EHOje kITCay"
    placeholder="Enter your login"
    value={null}
  >
    <label
      className="sc-bxivhb RKnZz"
    >
      <span>
        Login
      </span>
      <input
        className="sc-htpNat cjvsVh"
        onChange={[Function]}
        placeholder="Enter your login"
        type="text"
        value={null}
      />
    </label>
  </div>
  <div
    className="sc-EHOje kITCay"
    placeholder="Enter your password"
    value={null}
  >
    <label
      className="sc-bxivhb RKnZz"
    >
      <span>
        Password
      </span>
      <input
        className="sc-htpNat cjvsVh"
        onChange={[Function]}
        placeholder="Enter your password"
        type="password"
        value={null}
      />
    </label>
  </div>
  <div
    className="sc-htoDjs hyjxGL"
  >
    <button
      className="sc-bdVaJa cbeqcr sc-bwzfXH eydXok"
      onClick={[Function]}
    >
      Log in
    </button>
  </div>
</div>
`;
